cdap-proto/src/main/java/io/cdap/cdap/proto/sourcecontrol/SetRepositoryResponse.java

```diff
@@ -33,7 +33,7 @@ public SetRepositoryResponse(RepositoryConfigValidationException e) {
   }
 
   public SetRepositoryResponse(String message) {
-    this.errors = Collections.unmodifiableList(new ArrayList<>());
+    this.errors = null;
```

Comment:
Collections.emptylist() ?

Reply:
this is one of those rare cases where we might want null, so that it doesn't show up in the response json.

But then we should do the same for the other constructor (check if e.getFailures() is null or empty) and also remove the getter methods, as there should be no reason to ever use them.

Reply:
Added checking for `e.getFailures()`